src/components/Posts/PostMenu.tsx

```diff
@@ -0,0 +1,130 @@
+  Menu,
+  MenuItem,
+} from "@mui/material";
+import React, { useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import { MoreVert, Edit, Delete, Link } from "@mui/icons-material";
+import axios from "axios";
+import { deletePost } from "../../state/slices/postsSlice";
+import { useNavigate } from "react-router-dom";
+
+type PostMenuProps = {
+  authorId: number;
+  postId: number;
+  isExpanded?: boolean;
+};
+
+const styles = {
+  listItemIcon: {
+    "&.MuiListItemIcon-root": { color: "black.main" },
+  },
+  listItemText: { fontWeight: "bold" },
+  menu: {
+    borderRadius: 4,
+  },
+};
+
+const PostMenu = ({ authorId, postId, isExpanded = false }: PostMenuProps) => {
+  const userId = useAppSelector((state) => state.user.userId);
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+  const [menuOpen, setMenuOpen] = useState(false);
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  const handleDelete = async () => {
+    try {
+      const result = await axios.delete(
+        `http://localhost:3001/api/posts/deletePost`,
+        {
+          data: {
+            postId: postId,
+            userId: userId,
+          },
+        }
+      );
+
+      isExpanded ? handleExpandedPostDelete() : handlePostDelete();
+      return result.data;
+    } catch (error) {
+      console.error("Failed to delete the post", error);
+    } finally {
+      handleMenuClose();
+    }
+  };
+
+  const handleMenuOpen = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+    setMenuOpen(true);
+  };
+
+  const handleMenuClose = () => {
+    setAnchorEl(null);
+    setMenuOpen(false);
+  };
+
+  const handlePostDelete = () => {
+    dispatch(deletePost(postId));
+  };
+
+  const handleExpandedPostDelete = () => {
+    navigate(-1);
+  };
+
+  const handleTemporary = () => {
+    handleMenuClose();
+  };
+
+  return (
+    <>
+      <IconButton onClick={handleMenuOpen}>
+        <MoreVert />
+      </IconButton>
+      <Menu
+        anchorEl={anchorEl}
+        open={menuOpen}
+        onClose={handleMenuClose}
+        PaperProps={{
+          sx: styles.menu,
+        }}
+      >
+        {userId === authorId && [
+          <MenuItem onClick={handleTemporary} key="edit">
+            <ListItemIcon sx={styles.listItemIcon}>
+              <Edit />
+            </ListItemIcon>
+            <ListItemText primaryTypographyProps={styles.listItemText}>
+              Edit Post
+            </ListItemText>
+          </MenuItem>,
+          <MenuItem onClick={handleDelete} key="delete">
+            <ListItemIcon>
+              <Delete color="error" />
+            </ListItemIcon>
+            <ListItemText
+              primaryTypographyProps={{
+                ...styles.listItemText,
+                color: "error",
+              }}
+            >
+              Delete Post
+            </ListItemText>
+          </MenuItem>,
+        ]}
```

Comment:
There's some weird syntax going on here with array stuff lol, can you change it back to normal? The entire thing after `userId === authorId` is wrapped in square brackets

Reply:
Since Menu doesn't like using fragments I tried using an array instead for the two items only show if its the current user's post. I will make it two separate conditional renders for the two items if we do not want to use the array

Reply:
Ohh I see, I didn't know that the menu doesn't let us use fragments. I'm ok with this, if we want to get rid of the multiple conditional renders we can wrap it in a `Box` too, but I'm fine either way